### PR TITLE
[AIRFLOW-797] Kombu sqlachemy support

### PR DESCRIPTION
**Pull request by aminghadersohi**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- *(https://issues.apache.org/jira/browse/AIRFLOW-797)*

Per Apache guidelines you need to create a [Jira issue](https://issues.apache.org/jira/browse/AIRFLOW/).

### CODE REVIEW

**Comment by aminghadersohi** [comment]
@artwr this fixes an issue that prevents Airflow from running out of the box as the default config uses sqla+mysql for celery queue and results backend and the latest celery uses Kombu 4 that removed support for this. See Jira ticket.

File: airflow/executors/celery_executor.py [hunk 1]
```diff
@@ -43,6 +43,10 @@ class CeleryConfig(object):
     CELERYD_CONCURRENCY = configuration.getint('celery', 'CELERYD_CONCURRENCY')
     CELERY_DEFAULT_QUEUE = DEFAULT_QUEUE
     CELERY_DEFAULT_EXCHANGE = DEFAULT_QUEUE
+    if BROKER_URL.startswith('sqla') or CELERY_RESULT_BACKEND.startswith('sqla'):
```
Comment (artwr) [inline airflow/executors/celery_executor.py]: Shouldn't this be `BROKER_URL` only? `CELERY_RESULT_BACKEND` is usually `db+` from what I read in the [old docs](http://docs.celeryproject.org/en/3.1/configuration.html#conf-database-result-backend)

File: setup.py [hunk 2]
```diff
@@ -106,7 +106,8 @@ def write_version(filename=os.path.join(*['airflow',
 ]
 celery = [
     'celery>=3.1.17',
-    'flower>=0.7.3'
+    'flower>=0.7.3',
+    'karellen-kombu-ext==4.0.2'
```
Comment (artwr) [inline setup.py:110]: While this is a temporary solution, we also need to make sure that the dependencies we introduce are maintained. I really hope that your PR against kombu gets approved.

**Review by artwr — COMMENTED** [review]
You probably need to remove the condition on the RESULT_BACKEND as I do no think it was ever a possibility.

The new dependency does not look like an issue at first, but I am worried of introducing something without knowing the level of commitment, especially going forward.

File: setup.py (hunk 2 above)
Comment (aminghadersohi) [reply setup.py:110]: I agree. I will work with the maintainers to Kombu to do whatever it take to get it in.

File: airflow/executors/celery_executor.py (hunk 1 above)
Comment (aminghadersohi) [reply airflow/executors/celery_executor.py]: good catch

**Comment by aminghadersohi** [comment]
PR against kombu https://github.com/celery/kombu/pull/687

**Comment by aminghadersohi** [comment]
PR against kombu celery/kombu#687 is merged! 💃

**Pull request closed without merging**